app/modules/Home/HomeScreen.tsx

```diff
@@ -1,10 +1,28 @@
-import React, { FC } from 'react';
+import { useFocusEffect } from '@react-navigation/native';
+import React, { FC, useCallback } from 'react';
 import { Text, View } from 'react-native';
+import { HomeActions, useAppDispatch } from '../../redux';
 
 const HomeScreen: FC = () => {
+  const dispatch = useAppDispatch();
+  const fetchCompanyDetailsData = useCallback(() => {
+    dispatch(
+      HomeActions.getTrendingData({
+        type: 'all',
+        time: 'day',
+      }),
+    );
+  }, [dispatch]);
+
+  useFocusEffect(
+    useCallback(() => {
+      fetchCompanyDetailsData();
```

Comment:
here I do not see any other were to use fetchCompanyDetailsData function and not TODO comment, so please merge useFocusEffect and useCallback hooks.